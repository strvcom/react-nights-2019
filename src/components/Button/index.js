import styled from 'styled-components/macro'
import theme from '../../common/theme'

const Button = styled.button`
  background: ${theme.color.red};
  padding: 1rem;
  margin-top: 0.5rem;
  border: none;
  border-radius: ${theme.radius.basic};
  color: ${theme.color.white};
`

export default Button
