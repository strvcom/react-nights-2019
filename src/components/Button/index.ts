import styled from 'styled-components'
import theme from '../../common/theme'

const Button = styled.button`
  background: ${({ disabled }) =>
    disabled ? theme.color.gray : theme.color.red};
  cursor: ${({ disabled }) => (disabled ? 'default' : 'pointer')};
  padding: 1rem;
  margin-top: 0.5rem;
  border: none;
  border-radius: ${theme.radius.basic};
  color: ${theme.color.white};
`

export default Button
