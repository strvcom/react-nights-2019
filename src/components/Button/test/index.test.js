import React from 'react'
import { render } from 'react-testing-library'
import 'jest-styled-components'

import Button from '../index'

describe('[components] Button', () => {
  it('should render correctly', () => {
    const renderer = render(<Button>My Button</Button>)
    expect(renderer.baseElement).toMatchSnapshot()
  })
})
