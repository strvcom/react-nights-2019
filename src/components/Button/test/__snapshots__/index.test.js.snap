// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[components] Button should render correctly 1`] = `
.c0 {
  background: #ef0d33;
  cursor: pointer;
  padding: 1rem;
  margin-top: 0.5rem;
  border: none;
  border-radius: 5px;
  color: #fff;
}

<div>
  <button
    class="c0"
  >
    My Button
  </button>
</div>
`;

exports[`[components] Button when disabled should render correctly 1`] = `
.c0 {
  background: #e5e5e5;
  cursor: default;
  padding: 1rem;
  margin-top: 0.5rem;
  border: none;
  border-radius: 5px;
  color: #fff;
}

<div>
  <button
    class="c0"
    disabled=""
  >
    My Button
  </button>
</div>
`;
